### PR TITLE
Feature/kdg 23 register page

### PR DESCRIPTION
**Pull request by antonv-elision**

### CODE REVIEW

**Comment by antonv-elision** [comment]
image
image
image
image

File: marketplace/src/app/components/login/login.component.ts [hunk 1]
```diff
@@ -1,11 +1,36 @@
...
+  });
+  registerForm = new FormGroup({
+    mail: new FormControl('', Validators.email),
+  });
+
+  constructor(private router: Router) { }
+
   ngOnInit(): void {
-    // ngOnInit is a required function
+    // ngOnInit is required
```
Comment (GeertFeytons) [inline marketplace/src/app/components/login/login.component.ts]: This is not required if you remove the implement of the OnInit Interface

File: marketplace/src/app/components/login/login.component.ts [hunk 2]
```diff
@@ -1,11 +1,36 @@
...
+  }
+
+  onLogin(): void {
+    console.log(this.loginForm)
```
Comment (GeertFeytons) [inline marketplace/src/app/components/login/login.component.ts]: Remove console log

File: marketplace/src/app/components/login/login.component.ts [hunk 3]
```diff
@@ -1,11 +1,36 @@
...
+  }
+
+  onRegister(): void {
+    console.log(this.registerForm)
```
Comment (GeertFeytons) [inline marketplace/src/app/components/login/login.component.ts]: Remove console log

File: marketplace/src/app/components/login/login.component.ts [hunk 4]
```diff
@@ -1,11 +1,36 @@
...
+import getMailHint from "../../helpers/getMailHint";
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html'
 })
 export class LoginComponent implements OnInit {
+  public getMailHint = getMailHint;
+
+  loginForm = new FormGroup({
```
Comment (GeertFeytons) [inline marketplace/src/app/components/login/login.component.ts:13]: Shouldn't these form fields be required ?

File: marketplace/src/app/app.module.ts [hunk 5]
```diff
@@ -18,6 +18,11 @@ import {BadgeModule} from "primeng/badge";
...
 import {BrowserAnimationsModule} from "@angular/platform-browser/animations";
 import { LoginComponent } from './components/login/login.component';
+import {RippleModule} from "primeng/ripple";
+import {PasswordModule} from "primeng/password";
+import {ReactiveFormsModule} from "@angular/forms";
+import { RegisterComponent } from './components/register/register.component';
+import {DividerModule} from "primeng/divider";
 
 @NgModule({
   declarations: [
```
Comment (GeertFeytons) [inline marketplace/src/app/app.module.ts:28]: You are declaring every component in the AppModule but it's better the create separte modules that bundles certain components. This way it keeps you AppModule cleaner

File: marketplace/src/app/components/register/register.component.html [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        <h4>Personal details</h4>
+        <label for="firstName" class="block text-900 text-xl font-medium mb-2">First name</label>
+        <input id="firstName" type="text" placeholder="First name" pInputText class="w-full mb-3" formControlName="firstname" required minlength="2" autofocus>
+        <label for="familyName" class="block text-900 text-xl font-medium mb-2">Family name</label>
+        <input id="familyName" type="text" placeholder="Family name" pInputText class="w-full mb-3" formControlName="familyname" required minlength="2">
+        <label for="email2" class="block text-900 text-xl font-medium mb-2">Email</label>
+        <input id="email2" type="email" placeholder="Email address" pInputText class="w-full" formControlName="mail" required
+               (keydown.Tab)="getMailHint(registerForm.get('mail').value) && registerForm.get('mail').setValue(getMailHint(registerForm.get('mail').value))">
+        <p [hidden]="!getMailHint(registerForm.get('mail').value)">
+          Do you mean <a (click)="registerForm.get('mail').setValue(getMailHint(registerForm.get('mail').value))">{{getMailHint(registerForm.get('mail').value)}}</a>?
```
Comment (GeertFeytons) [inline marketplace/src/app/components/register/register.component.html]: It's better to use a button and style it so it looks like an "a" then using an "a" that functions as a button

File: marketplace/src/app/helpers/getMailHint.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+export default function getMailHint(input: string) {
+  const domains = ['gmail.com', 'yahoo.com', 'hotmail.com', 'aol.com', 'elision.eu', 'cronos.be', 'hotmail.be', 'msn.com', 'live.com', 'icloud.com', 'outlook.com']
+  if(!input.includes('@')) return false
+  const parts = input.split('@', 2)
+  if(parts[1].length === 0 || parts[1].includes('.')) return false
+  for(let domain of domains) {
```
Comment (GeertFeytons) [inline marketplace/src/app/helpers/getMailHint.ts]: You can also use domains.find(domain => domain.startsWith(parts[1])) here

**Review by GeertFeytons — CHANGES_REQUESTED** [review]

File: marketplace/src/app/components/login/login.component.ts (hunk 4 above)
Comment (antonv-elision) [reply marketplace/src/app/components/login/login.component.ts:13]: Login is another ticket, this is simply a form template that is included in the page

File: marketplace/src/app/app.module.ts (hunk 5 above)
Comment (antonv-elision) [reply marketplace/src/app/app.module.ts:28]: Will solve this in a seperate issue

**Review by schrogl — APPROVED** [review]

**Pull request merged**